db/view/view.cc

```diff
@@ -318,6 +320,50 @@ static bool is_partition_key_empty(
     }
 }
 
+// Checks if the result matches the provided view filter
+class view_filter_checking_visitor {
+    const schema& _base;
+    const view_info& _view;
+    const std::vector<bytes>& _pk;
+    const std::vector<bytes>& _ck;
+
+    bool _matches_view_filter = true;
+public:
+    view_filter_checking_visitor(const schema& base, const view_info& view, const std::vector<bytes>& pk, const std::vector<bytes>& ck)
+        : _base(base)
+        , _view(view)
+        , _pk(pk)
+        , _ck(ck)
+    {}
+
+    void accept_new_partition(const partition_key& key, uint64_t row_count) {}
+    void accept_new_partition(uint64_t row_count) {}
+
+    void accept_new_row(const clustering_key& key, const query::result_row_view& static_row, const query::result_row_view& row) {
+        _matches_view_filter &= check_if_matches(static_row, row);
+    }
+
+    void accept_new_row(const query::result_row_view& static_row, const query::result_row_view& row) {
+        _matches_view_filter &= check_if_matches(static_row, row);
+    }
+    void accept_partition_end(const query::result_row_view& static_row) {}
+
+    bool check_if_matches(const query::result_row_view& static_row, const query::result_row_view& row) const {
+        auto selection = cql3::selection::selection::wildcard(_base.shared_from_this());
+        return boost::algorithm::all_of(
+            _view.select_statement().get_restrictions()->get_non_pk_restriction() | boost::adaptors::map_values,
+            [&] (auto&& r) {
+                return cql3::expr::is_satisfied_by(
+                        r->expression, _pk, _ck, static_row, &row, *selection, cql3::query_options({ }));
```

Comment:
Later, we can have just a single expression object representing all the filters.

Reply:
Yes, that's inevitable since the type returned by get_non_pk_restrictions() will be deleted.  It will naturally fall to me to do it.